aptos-move/framework/src/natives/cryptography/ristretto255_point.rs

```diff
@@ -502,6 +522,38 @@ pub(crate) fn native_new_point_from_64_uniform_bytes(
     Ok(NativeResult::ok(cost, smallvec![Value::u64(result_handle)]))
 }
 
+pub(crate) fn native_double_scalar_mul(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    mut _ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(args.len(), 4);
+
+    let point_context = context.extensions().get::<NativeRistrettoPointContext>();
+    let mut point_data = point_context.point_data.borrow_mut();
+
+    let scalar2 = pop_scalar_from_bytes(&mut args)?;
+    let scalar1 = pop_scalar_from_bytes(&mut args)?;
+    let handle2 = pop_as_ristretto_handle(&mut args)?;
+    let handle1 = pop_as_ristretto_handle(&mut args)?;
+
+    let points = vec![
+        point_data.get_point(&handle1),
+        point_data.get_point(&handle2),
+    ];
+
+    let scalars = vec![scalar1, scalar2];
+
+    let result = RistrettoPoint::vartime_multiscalar_mul(scalars.iter(), points.into_iter());
+
+    let cost = gas_params.multi_scalar_mul_gas(2);
```

Comment:
TODO: just give a discount here, instead of using the formula

Reply:
What does this mean?

Reply:
I meant instead of using the gas formula in `multi_scalar_mul_gas(num_exps)`, just charge `2 * individual_exp_cost / discount`. Benchmark 2 `scalar_mul`'s versus `double_scalar_mul` to figure out what `discount` should be.